diff-tests/core/driver/setup.go

```diff
@@ -0,0 +1,761 @@
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/interchain-security/x/ccv/types"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/ibc-go/v3/testing/mock"
+
+	ibctesting "github.com/cosmos/ibc-go/v3/testing"
+
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	simapp "github.com/cosmos/interchain-security/testutil/simapp"
+
+	cryptoEd25519 "crypto/ed25519"
+
+	cosmosEd25519 "github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v3/modules/core/23-commitment/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v3/modules/light-clients/07-tendermint/types"
+
+	slashingkeeper "github.com/cosmos/cosmos-sdk/x/slashing/keeper"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	appConsumer "github.com/cosmos/interchain-security/app/consumer"
+	appProvider "github.com/cosmos/interchain-security/app/provider"
+	simibc "github.com/cosmos/interchain-security/testutil/simibc"
+	consumerkeeper "github.com/cosmos/interchain-security/x/ccv/consumer/keeper"
+	consumertypes "github.com/cosmos/interchain-security/x/ccv/consumer/types"
+	providerkeeper "github.com/cosmos/interchain-security/x/ccv/provider/keeper"
+
+	channelkeeper "github.com/cosmos/ibc-go/v3/modules/core/04-channel/keeper"
+	ccv "github.com/cosmos/interchain-security/x/ccv/types"
+)
+
+type Builder struct {
+	suite          *suite.Suite
+	link           simibc.OrderedLink
+	path           *ibctesting.Path
+	coordinator    *ibctesting.Coordinator
+	clientHeaders  map[string][]*ibctmtypes.Header
+	mustBeginBlock map[string]bool
+	valAddresses   []sdk.ValAddress
+	initState      InitState
+}
+
+func (b *Builder) ctx(chain string) sdk.Context {
+	return b.chain(chain).GetContext()
+}
+
+func (b *Builder) chainID(chain string) string {
+	return map[string]string{P: ibctesting.GetChainID(0), C: ibctesting.GetChainID(1)}[chain]
+}
+
+func (b *Builder) otherID(chainID string) string {
+	return map[string]string{ibctesting.GetChainID(0): ibctesting.GetChainID(1), ibctesting.GetChainID(1): ibctesting.GetChainID(0)}[chainID]
+}
+
+func (b *Builder) chain(chain string) *ibctesting.TestChain {
+	return map[string]*ibctesting.TestChain{P: b.providerChain(), C: b.consumerChain()}[chain]
+}
+
+func (b *Builder) providerChain() *ibctesting.TestChain {
```

Comment:
I think so. I would prefer though to merge as is and track these improvements in new issues. The setup.go is particularly low quality, but it will take some time to improve it because it uses the ibctesting.Coordinator

Reply:
Yea this will become an issue similar to https://github.com/cosmos/interchain-security/issues/300, but agreed that it may not be the most important thing to focus on rn 👍